### PR TITLE
Rabin Karp Search Algorithm

### PR DESCRIPTION
**Pull request by aladin002dz**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in their comments that points to Wikipedia or another similar explanation.
- [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: Search/RabinKarp.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    // Check if the hashes match and perform a character-wise comparison
+    if (hashText[i] === hashPattern) {
+      if (text.substring(i, i + patternLength) === pattern) {
+        indices.push(i) // Store the index where the pattern is found
+      }
+    }
+  }
+
+  return indices.length > 0 ? indices : [-1] // Return an array of indices where the pattern is found or [-1] if not found
```
Comment (appgurueu) [inline Search/RabinKarp.js:60]: Returning an empty array would be more sensible.

File: Search/RabinKarp.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * The algorithm returns an array of indices where the pattern is found in the text. If the pattern is not
+ * found, the algorithm returns an empty array.
+ *
+ * [Reference](https://en.wikipedia.org/wiki/Rabin%E2%80%93Karp_algorithm)
+ */
+
+function hash(str, length) {
+  let hashValue = 0
+  for (let i = 0; i < length; i++) {
+    hashValue = (hashValue * 256 + str.charCodeAt(i)) % 997
```
Comment (appgurueu) [inline Search/RabinKarp.js:19]: You should use the `BASE` and `MOD` constants here. For that purpose, you should move them to outside the `hash` function.

File: Search/RabinKarp.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
+/*
+ * Implements the Rabin-Karp algorithm for pattern searching.
```
Comment (appgurueu) [inline Search/RabinKarp.js:2]: This comment should be moved to the `rabinKarpSearch` function.

**Review by appgurueu — COMMENTED** [review]
Otherwise looks fine. (I see some of the PRs we merged weren't properly formatted, our bad...)

**Review by appgurueu — DISMISSED** [review]

**Review by appgurueu — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**